### PR TITLE
build: allow to build on FreeBSD

### PR DESCRIPTION
**Pull request by gearnode**
Fix FreeBSD build. All is explain in the commit message.

ref  #16902

### CODE REVIEW

File: scripts/fetch-ui-assets.sh [hunk 1]
```diff
@@ -26,9 +26,16 @@ curl -Ls https://github.com/influxdata/ui/releases/download/OSS-Master/sha256.tx
 # Download the tar file containing the built UI assets.
 curl -L https://github.com/influxdata/ui/releases/download/OSS-Master/build.tar.gz --output build.tar.gz
 
+
 # Verify the checksums match; exit if they don't.
-echo "$(cat sha256.txt)" | sha256sum --check -- \
-    || { echo "Checksums did not match for downloaded UI assets!"; exit 1; }
+if [ "$(uname)" = "FreeBSD" ]; then
```
Comment (ayang64) [inline scripts/fetch-ui-assets.sh:31]: thank you -- i really appreciate you driving this home!  this is purely a suggestion and i don't feel strongly about this:

what do you think about using a `case` block for detecting the operating system?  something like:

```
case $(uname -s) in
   FreeBSD)
    ...
   Darwin)
    ...
   Linux)
    ...
    *)
      ...
esac

**Review by ayang64 — APPROVED** [review]
perfect!  thank you!

File: scripts/fetch-ui-assets.sh (hunk 1 above)
Comment (gearnode) [reply scripts/fetch-ui-assets.sh:31]: @ayang64 I will be happy to refactor the code :) But while no maintainer interacts with the pull request, I don't want to lose my time on this...

Comment (danxmoran) [reply scripts/fetch-ui-assets.sh:31]: 👍 to @ayang64's suggestion. `sha256sum` is non-standard on Mac as well, so I'd suggest:
```
case $(uname -s) in
    FreeBSD|Darwin)
        # Use shasum --algorithm 256
    *)
        # Use sha256sum
esac
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -84,6 +84,7 @@ This release adds an embedded SQLite database for storing metadata required by t
 1. [#22211](https://github.com/influxdata/influxdb/pull/22211): Prevent scheduling an inactivated tasks after updating it
 1. [#22235](https://github.com/influxdata/influxdb/pull/22235): Avoid compaction queue stats flutter
 1. [#22272](https://github.com/influxdata/influxdb/pull/22272): Requests to `/api/v2/authorizations` filter correctly on `org` and `user` parameters
+1. [#22285](https://github.com/influxdata/influxdb/pull/22285): Fix FreeBSD build.
```
Comment (danxmoran) [inline CHANGELOG.md:87]: We just set up some automation for changelog-ing, so you can revert this line

**Comment by gearnode** [comment]
@danxmoran Patches updated :)

**Comment by danxmoran** [comment]
@gearnode didn't notice this last time: is renaming `Makefile` to `GNUmakefile` a strict requirement? It looks like CI is happy with it but I worry about unintended side-effects...

**Comment by ayang64** [comment]
> @gearnode didn't notice this last time: is renaming `Makefile` to `GNUmakefile` a strict requirement? It looks like CI is happy with it but I worry about unintended side-effects...

renaming it `GNUmakefile` would be nice if it has any GNU extensions (which, IIRC, it does).  FreeBSD doesn't come with GNU make out of the box -- it uses the original `pmake` which isn't compatible.

imho, it is best to indicate that we're using GNU extensions in the make file by naming it appropriately.

**Comment by gearnode** [comment]
> @gearnode didn't notice this last time: is renaming `Makefile` to `GNUmakefile` a strict requirement? It looks like CI is happy with it but I worry about unintended side-effects...

Renaming the file has no side effect as documented by the GNU make(1) man page:

> By default, when make looks for the makefile, it tries the following names, in order: GNUmakefile, makefile and Makefile. 
> ...
>  You should use this name if you have a makefile that is specific to GNU make, and will not be understood by other versions of make.

The filename is a way to indicate that the Makefile is only compatible with the GNU version of make.

**Comment by gearnode** [comment]
@danxmoran any update?

**Comment by galdor** [comment]
As someone using FreeBSD, this would be really useful to me, I'm trying to switch to Influx.

File: scripts/fetch-ui-assets.sh [hunk 3]
```diff
@@ -27,8 +27,17 @@ curl -Ls https://github.com/influxdata/ui/releases/download/OSS-Master/sha256.tx
...
-    || { echo "Checksums did not match for downloaded UI assets!"; exit 1; }
+case "$(uname -s)" in
+    FreeBSD | Darwin)
+        echo "$(cat sha256.txt)" | shasum --algorithm 256 --check \
+            || { echo "Checksums did not match for downloaded UI assets!"; exit 1; } ;;
+    Linux)
+        echo "$(cat sha256.txt)" | sha256sum --check -- \
+            || { echo "Checksums did not match for downloaded UI assets!"; exit 1; } ;;
+    *)
+        echo "The '$(uname -s)' operating system is not supported" >&2
```
Comment (lesam) [inline scripts/fetch-ui-assets.sh:38]:
```suggestion
        echo "The '$(uname -s)' operating system is not supported as a build host for the UI" >&2
```

**Review by lesam — COMMENTED** [review]
LGTM with the message change

**Review by lesam — APPROVED** [review]

**Comment by lesam** [comment]
I'll make the message change separately

**Pull request merged**